README.md
# homebridge-dashboard

This is a collaborative project of [@sschuste](https://github.com/sschuste) and [@nastras](https://github.com/nastras).

## What is homebridge-dashboard?

A welcome screen for homebridge called based on a bash script so this is not a regular homebridge-plugin. 
It is an additional tool for admins and users that displays an overview of important system and homebridge information in the shell.


## What does homebridge-dashboard offer you?

### Display of:
- System data
- NPM updates
- Homebridge, Node, NPM version
- Status of homebridge instances
- Status of deconz networks (See also Display of additional modules)

![example-homebridge-dashboard-screen](example-homebridge-dashboard.png)

## Notes: 
- The script needs npm-check installed: (sudo) npm install -g npm-check
- The script was developed on a Raspberry Pi and I'm pretty sure that it won't work on an other OS than Raspberian. 
- And, even more important: this is work in progress.
- Version changes can be viewed in the [Changelog](https://github.com/Nastras/homebridge-dashboard/blob/master/Changelog.md).
```diff
! Feedback and improvements are welcome !
```

## Installation
Copy & paste this into your console:
```
wget -qO - https://raw.githubusercontent.com/nastras/homebridge-dashboard/master/dashboard-installer.sh | bash
```

The installer installs the script to */usr/local/bin/homebridge-dashboard.sh* and the language files to */usr/local/share/homebridge-dashboard*. To do so you need root access or sudo privileges on your computer.

If you don't have these rights or you don't want the files to be installed in these directories, you can install the dashboard to your home directory by cloning the git repository:
```
git clone https://github.com/nastras/homebridge-dashboard.git
```
This creates a directory named *homebridge-dashboard* in the current directory.

## Usage
```
homebridge-dashboard.sh [-p path][-l language]
```
If you have installed the standard installation as described above you only have to run homebridge-dashboard without any further arguments. The script takes some seconds to run, so please be patient.

## Language support
The script tries to determine automatically what language your computer is running on. To change the language you have to pass the language argument on the command line:
```
homebridge-dashboard.sh -l <language>
```

To switch to another language, run
```
homebridge-dashboard.sh -l en_gb.utf-8
```
or
```
homebridge-dashboard.sh --lang=en_gb.utf-8
```

Currently only english and german is supported. Feel free to add any other language by adding a new language file. See below how to do this.

| Language  | Argument    | Comment                         |
| --------- | ----------- | ------------------------------- |
| English   | en_gb.utf-8 |                                 |
| American  | en_us.utf-8 | Shows temperature in Fahrenheit |
| German    | de_de.utf-8 |                                 |

## Change the path to the language files
If your language files are located on another place than /usr/local/share/homebridge-dashboard you have to run homebridge-dashboard.sh with the path argument:

```
homebridge-dashboard.sh -p <path>
```
For instance:
```
homebridge-dashboard.sh -p /home/pi/homebridge-dashboard/i18n
```
or 
```
homebridge-dashboard.sh --path=/home/pi/homebridge-dashboard/i18n
```

## Writing language files
First you have to determine the language of your computer with:
```
locale | grep LANG=
```

You'll get something like LANG=fr_FR.UTF-8 if your computer is set to french, or maybe LANG=ru_RU.utf-8 if it's running in russian language. The part right from the equal sign in lower characters is the file name of your new language file, i.e. fr_fr.utf-8.

Copy the english language file to your new file name:
```
cp /usr/local/share/homebridge-dashboard/en_gb.utf-8 fr_fr.utf-8
```
Then edit the new file:
```
nano fr_fr.utf-8
```
and do your translation. When you're done copy the file to your language directory:
```
sudo cp fr_fr.utf-8 /usr/local/share/homebridge-dashboard/
```

and run homebridge-dashboard.sh without arguments or as:
```
homebridge-dashboard.sh --lang=fr_fr.utf-8
```

## Automatic start on login

Customize /home/pi/.profile
The path of the script we have to insert in the /home/pi/.profile in the last line.

Example: /usr/local/bin/homebridge-dashboard.sh
```
sudo nano /home/pi/.profile
```

## Display of additional modules

Open the script homebridge-dashboard.sh to edit. At the end of the script the respective module is activated by removing the # symbols. Hint: currently there is only deconz networks as additional module.

Example: to activate the deconz module remove the # at the beginning of the line an save:
- 194
- 197
- 224
- 225 

![example-activate-modules](example-activate-modules.png)

## Removing the default message's on login

### Message 1
Remove the contents of the file and save.
```
sudo nano /etc/motd
```

### Message 2
Open file and comment out second line uname -snrvm with # and save.

Example: #uname -snrvm
```
sudo nano /etc/update-motd.d/10-uname
```

### Message 3
Search and change line PrintLastLog yes in PrintLastLog no and save.
```
sudo nano /etc/ssh/sshd_config
```


Changelog.md
# Changelog

## Version 2.0.0 - 2019-09-22
- Script completely translated into English.
- Changed the order of the displayed lines.
- Display Colors assigned by topic, no longer by traffic light.

- Homebridge total added. Now you can see how many instances you have completely, how many are active and how many have stopped or failed.
- Homebridge area extended in script. So far, inactive instances are only displayed if they have been in restart mode or activating mode. Now also the instances that were stopped manually are displayed.

- Optional display of deCONZ networks built in. See also in readme: Display of additional modules.

## Version 1.4.2 - 2019-08-25
- Bug Fix: [@Issue 5](https://github.com/Nastras/homebridge-dashboard/issues/5)

## Version 1.4.1 - 2019-08-12
- Bug Fix: Under certain circumstances, the number of node_modules updates was displayed incorrectly.

## Version 1.4.0 - 2019-08-04
- Multilanguage support for: 

(de_de.utf-8)

(en_gb.utf-8)

(en_us.utf-8)

- Automatic installation with dashboard-installer.sh


Archiv/v.1.4.2-homebridge-dashboard.sh
#!/bin/bash
# Name: homebridge-dashboard.sh
# Version: 1.4.2
# Autor: sschuste & Nastra (SmartApfel Forum)
# Link: https://github.com/Nastras/homebridge-dashboard

TEMP=`getopt -o l:p: --long lang:,path: -- "$@"`

if [ $? != 0 ] ; then echo "Terminating..." >&2 ; exit 1 ; fi

eval set -- "$TEMP"

while true; do
	case "$1" in
		-l | --lang ) I18N_FILE=$2; shift 2 ;;
		-p | --path ) I18N_PATH=$2; shift 2 ;;
		-- ) shift; break ;;
		* ) break ;;
	esac
done

# Did we get the I18N_FILE via a command line argument? If not, then grep it from locale.
if [ "x$I18N_FILE" == "x" ]; then
	I18N_FILE=$(locale | grep "LC_CTYPE" | awk -F= '{print $2}' | tr '[:upper:]' '[:lower:]' | tr -d \")

	# If we still don't know the language, set it to english
  	if [ "x$I18N_FILE" == "x" ]; then
		I18N_FILE="en_gb.utf-8"
	fi
fi

# Did we get the path to the i18n file via a command line argument? 
if [ ! -z $I18N_PATH ]; then 
	# Yes, we got a path as command line argument
	I18N_PATH="$I18N_PATH/"
	if [ ! -f "$I18N_PATH$I18N_FILE" ]; then
		echo "Couldn't find $I18N_PATH$I18N_FILE. Exiting."
		exit 1
	fi
else
	# No, there was no path in command line argument
	DEFAULT_PATH="/usr/local/share/homebridge-dashboard/"
	if [ ! -f "$DEFAULT_PATH$I18N_FILE" ]; then
		echo "Couldn't find $DEFAULT_PATH$I18N_FILE. Exiting."
		exit 1
	else
		I18N_PATH=$DEFAULT_PATH
	fi
fi

. "$I18N_PATH$I18N_FILE" 
export POSIXLY_CORRECT=1

function output {
	case "$1" in
		red)
		COLOR="\e[1;31m"
		;;
		green)
		COLOR="\e[1;32m"
		;;
		yellow)
		COLOR="\e[1;33m"
		;;
		magenta)
		COLOR="\e[1;35m"
		;;
		cyan)
		COLOR="\e[1;36m"
		;;
		white)
		COLOR="\e[1;97m"
		;;
	esac

	let DOTS=21-$(echo -n $2 | wc -m)
        FILL=$(echo $(for i in $(seq 1 $DOTS); do printf "."; done))
        echo -e "\e[0;35m$4\e[0;97m$2$FILL: $COLOR$3"
}


# Datum
DATUM=`date +"%A, %e %B %Y"`

# Zeit
ZEIT=`date +"%H:%M:%S %Z"`

# Modell
MODELL=`cat /sys/firmware/devicetree/base/model | tr -d '\0'`

# System
SYSTEM=`hostnamectl | grep System | awk -F: '{print $2}' | xargs`

# Kernel
KERNEL=`uname -rs`

# Uptime
UP0=`cut -d. -f1 /proc/uptime`
UP1=$(($UP0/86400))		# Tage
UP2=$(($UP0/3600%24))	# Stunden
UP3=$(($UP0/60%60))		# Minuten
UP4=$(($UP0%60))		# Sekunden
UPTIME="$UP1 $DISPLAY_DAYS, $UP2:$UP3 $DISPLAY_HOURS"

# Durchschnittliche Auslasung
LOAD1=`cat /proc/loadavg | awk '{print $1}'`	# Letzte Minute
LOAD5=`cat /proc/loadavg | awk '{print $2}'`	# Letzte 5 Minuten
LOAD15=`cat /proc/loadavg | awk '{print $3}'`	# Letzte 15 Minuten
LOAD="$LOAD1 ($DISPLAY_LOAD1) | $LOAD5 ($DISPLAY_LOAD5) | $LOAD15 ($DISPLAY_LOAD15)"

# Arbeitsspeicher
RAM1=`free -h --mega | grep 'Mem' | awk '{print $2}'`	# Total
RAM2=`free -h --mega | grep 'Mem' | awk '{print $3}'`	# Used
RAM3=`free -h --mega | grep 'Mem' | awk '{print $4}'`	# Free
RAM4=`free -h --mega | grep 'Swap' | awk '{print $3}'`	# Swap used
RAM="$DISPLAY_RAMTOTAL: $RAM1 | $DISPLAY_RAMUSED: $RAM2 | $DISPLAY_RAMFREE: $RAM3 | $DISPLAY_RAMSWAP: $RAM4"

# Temperatur
TEMP=$(cat /sys/class/thermal/thermal_zone0/temp)
if [ "$DISPLAY_TEMPUNIT" == "C" ]; then
	TEMP=$(awk -v TEMP=$TEMP 'BEGIN { printf "%.1f",TEMP/1000 }')
fi
if [ "$DISPLAY_TEMPUNIT" == "F" ]; then
	TEMP=$(awk -v TEMP=$TEMP 'BEGIN { printf "%.1f",((9/5) * TEMP/1000) +32 }')
fi
TEMP="$TEMPº $DISPLAY_TEMPUNIT"


# Speicher verbunden
SPEICHERSD=`lsblk | grep -w mmcblk0 | grep -i disk | awk '{print $1}'`
SPEICHERUSB1=`lsblk | grep -w sda | grep -i disk | awk '{print $1}'`
SPEICHERUSB2=`lsblk | grep -w sdb | grep -i disk | awk '{print $1}'`
SPEICHERUSB3=`lsblk | grep -w sdc | grep -i disk | awk '{print $1}'`
SPEICHERUSB4=`lsblk | grep -w sdd | grep -i disk | awk '{print $1}'`
DISKS="SD: $SPEICHERSD | USB1: $SPEICHERUSB1 | USB2: $SPEICHERUSB2 | USB3: $SPEICHERUSB3 | USB4: $SPEICHERUSB4"

# Speicherbeleguwng
DISK1=`df -h | grep 'dev/root' | awk '{print $2}'`	# Gesamtspeicher
DISK2=`df -h | grep 'dev/root' | awk '{print $3}'`	# Belegt
DISK3=`df -h | grep 'dev/root' | awk '{print $4}'`	# Frei
CAPACITY="$DISPLAY_DISKTOTAL: $DISK1 | $DISPLAY_DISKUSED: $DISK2 | $DISPLAY_DISKFREE: $DISK3"

# Hostname
HOSTNAME=`hostname -f`

# IP-Adressen ermitteln
if ( /sbin/ifconfig | grep -q "eth0" ) ; then IP_LAN=`ifconfig eth0 | grep "inet" | cut -d ":" -f 2 | awk '{print $2}'` ; else IP_LAN="---" ; fi ;
if ( /sbin/ifconfig | grep -q "wlan0" ) ; then IP_WLAN=`ifconfig wlan0 | grep "inet" | cut -d ":" -f 2 | awk '{print $2}'` ; else IP_WLAN="---" ; fi ;
IP_ADDRESSES="$DISPLAY_IPADDRLAN: $IP_LAN | $DISPLAY_IPADDRWIFI: $IP_WLAN"

# Benutzer
BENUTZER=`whoami | awk '{print $1}'`

# letzter Login
LAST1=`last -2 -a | awk 'NR==2{print $3}'`	# Wochentag
LAST2=`last -2 -a | awk 'NR==2{print $5}'`	# Tag
LAST3=`last -2 -a | awk 'NR==2{print $4}'`	# Monat
LAST4=`last -2 -a | awk 'NR==2{print $6}'`	# Uhrzeit
LAST5=`last -2 -a | awk 'NR==2{print $10}'`	# Remote-Computer
LAST="$LAST1, $LAST2 $LAST3 $LAST4 $DISPLAY_FROM $LAST5"

# Homebridge Version
HOMEBRIDGE=`homebridge --version`

# Node Version
NODE=`node -v | tr -d [:alpha:]`

# Npm Version
NPM=`npm -v | tr -d [:alpha:]`

# Npm Updates
# NPMUPDATES=`npm outdated -g | grep global -i | wc -l`
NPMUPDATES=`npm-check -g | grep -i " to go from" | wc -l`

# Homebridge Instanzen Aktiv
INSTANZAKTIV="$(systemctl -t service | grep -w homebridge | wc -l) $DISPLAY_INSTANCE"

# Homebridge Instanzen Inaktiv
INSTANZINAKTIV="$(systemctl -t service | grep -w homebridge | grep -i activating | wc -l) $DISPLAY_INSTANCE"


echo -e "\e[1;35m$DISPLAY_TITLE"
echo 
output "yellow"  "$DISPLAY_DATE" "$DATUM"						"                   ██                   "
output "magenta" "$DISPLAY_TIME" "$ZEIT"						"                  ████                  "
output "magenta" "$DISPLAY_MODEL" "$MODELL"						"                ████████    ███████     "
output "magenta" "$DISPLAY_SYSTEM" "$SYSTEM"						"              ███      ██████    ██     "
output "magenta" "$DISPLAY_KERNEL" "$KERNEL"						"            ███           ███    ██     "
output "magenta" "$DISPLAY_UPTIME" "$UPTIME"						"          ███      ███     ██    ██     "
output "magenta" "$DISPLAY_LOAD" "$LOAD"						"        ███      ███ ███    █    ██     "
output "magenta" "$DISPLAY_RAMUSAGE" "$RAM"						"      ███      ███     ███       ██     "
output "magenta" "$DISPLAY_TEMP" "$TEMP"						"    ███      ███         ███      ██    "
output "magenta" "$DISPLAY_DISKS" "$DISKS"						"  ███      ███     ███     ███     ██   "
output "magenta" "$DISPLAY_DISKCAPACITY" "$CAPACITY"					" ███     ███     ███████     ███    ██  "
output "magenta" "$DISPLAY_HOSTNAME" "$HOSTNAME"					" ███   ███     ███     ███     ███      "
output "magenta" "$DISPLAY_IPADDR" "$IP_ADDRESSES"					"   █████     ███         ███    ███     "
output "magenta" "$DISPLAY_USERNAME" "$BENUTZER"					"           ███     ██      ███  ███     "
output "magenta" "$DISPLAY_LASTLOGIN" "$LAST"						"         ███    ████████     ████       "
output "magenta" "$DISPLAY_HOMEBRIDGEVERSION" "$HOMEBRIDGE"				"        ███   ███      ███              "
output "magenta" "$DISPLAY_NODEVERSION" "$NODE"						"          █████         ███             "
output "magenta" "$DISPLAY_NPMVERSION" "$NPM"						"                         ███            "
output "cyan"    "$DISPLAY_NPMUPDATES" "$NPMUPDATES $DISPLAY_NPMUPDATESAVAILABLE"	"                       ███              "
output "green"   "$DISPLAY_HOMEBRIDGEACTIVE" "$INSTANZAKTIV"				"                     ███                "
output "red"     "$DISPLAY_HOMEBRIDGEINACTIVE" "$INSTANZINAKTIV"			"                   ███                  "


homebridge-dashboard.sh
#!/bin/bash
# Name: homebridge-dashboard.sh
# Version: 2.0.0
# Autor: sschuste & Nastra (German SmartApfel HomeKit Forum)
# Link: https://github.com/Nastras/homebridge-dashboard


# Script Language
TEMP=`getopt -o l:p: --long lang:,path: -- "$@"`

if [ $? != 0 ] ; then echo "Terminating..." >&2 ; exit 1 ; fi

eval set -- "$TEMP"

while true; do
case "$1" in
-l | --lang ) I18N_FILE=$2; shift 2 ;;
-p | --path ) I18N_PATH=$2; shift 2 ;;
-- ) shift; break ;;
* ) break ;;
esac
done

## Did we get the I18N_FILE via a command line argument? If not, then grep it from locale.
if [ "x$I18N_FILE" == "x" ]; then
I18N_FILE=$(locale | grep "LC_CTYPE" | awk -F= '{print $2}' | tr '[:upper:]' '[:lower:]' | tr -d \")

## If we still don't know the language, set it to english
if [ "x$I18N_FILE" == "x" ]; then
I18N_FILE="en_gb.utf-8"
fi
fi

## Did we get the path to the i18n file via a command line argument? 
if [ ! -z $I18N_PATH ]; then 

## Yes, we got a path as command line argument
I18N_PATH="$I18N_PATH/"
if [ ! -f "$I18N_PATH$I18N_FILE" ]; then
echo "Couldn't find $I18N_PATH$I18N_FILE. Exiting."
exit 1
fi
else

## No, there was no path in command line argument
DEFAULT_PATH="/usr/local/share/homebridge-dashboard/"
if [ ! -f "$DEFAULT_PATH$I18N_FILE" ]; then
echo "Couldn't find $DEFAULT_PATH$I18N_FILE. Exiting."
exit 1
else
I18N_PATH=$DEFAULT_PATH
fi
fi

. "$I18N_PATH$I18N_FILE" 
export POSIXLY_CORRECT=1

## Text color
function output {
case "$1" in
red)
COLOR="\e[1;31m"
;;
blue)
COLOR="\e[1;34m"
;;
green)
COLOR="\e[1;32m"
;;
yellow)
COLOR="\e[1;33m"
;;
magenta)
COLOR="\e[1;35m"
;;
cyan)
COLOR="\e[1;36m"
;;
white)
COLOR="\e[1;97m"
;;
esac

let DOTS=21-$(echo -n $2 | wc -m)
FILL=$(echo $(for i in $(seq 1 $DOTS); do printf "."; done))
echo -e "\e[0;35m$4\e[0;97m$2$FILL: $COLOR$3"
}

# Date
DATE=`date +"%A, %e %B %Y"`

# Time
TIME=`date +"%H:%M:%S %Z"`

# Model
MODEL=`cat /sys/firmware/devicetree/base/model | tr -d '\0'`

# System
SYSTEM=`hostnamectl | grep System | awk -F: '{print $2}' | xargs`

# Kernel
KERNEL=`uname -rs`

# Uptime
UP0=`cut -d. -f1 /proc/uptime`
UP1=$(($UP0/86400))		# Days
UP2=$(($UP0/3600%24))	# Hours
UP3=$(($UP0/60%60))		# Minutes
UP4=$(($UP0%60))		# Seconds
UPTIME="$UP1 $DISPLAY_DAYS, $UP2:$UP3 $DISPLAY_HOURS"

# CPU usage
LOAD1=`cat /proc/loadavg | awk '{print $1}'`	# Last minutes
LOAD5=`cat /proc/loadavg | awk '{print $2}'`	# Last 5 minutes
LOAD15=`cat /proc/loadavg | awk '{print $3}'`	# Last 15 minutes
LOAD="$LOAD1 ($DISPLAY_LOAD1) | $LOAD5 ($DISPLAY_LOAD5) | $LOAD15 ($DISPLAY_LOAD15)"

# Main memory
RAM1=`free -h --mega | grep 'Mem' | awk '{print $2}'`	# Total
RAM2=`free -h --mega | grep 'Mem' | awk '{print $3}'`	# Used
RAM3=`free -h --mega | grep 'Mem' | awk '{print $4}'`	# Free
RAM4=`free -h --mega | grep 'Swap' | awk '{print $3}'`	# Swap used
RAM="$DISPLAY_RAMTOTAL: $RAM1 | $DISPLAY_RAMUSED: $RAM2 | $DISPLAY_RAMFREE: $RAM3 | $DISPLAY_RAMSWAP: $RAM4"

# Temperature
TEMP=$(cat /sys/class/thermal/thermal_zone0/temp)
if [ "$DISPLAY_TEMPUNIT" == "C" ]; then
	TEMP=$(awk -v TEMP=$TEMP 'BEGIN { printf "%.1f",TEMP/1000 }')
fi
if [ "$DISPLAY_TEMPUNIT" == "F" ]; then
	TEMP=$(awk -v TEMP=$TEMP 'BEGIN { printf "%.1f",((9/5) * TEMP/1000) +32 }')
fi
TEMP="$TEMPº $DISPLAY_TEMPUNIT"


# Connected hard drive
NAMESD=`lsblk | grep -w mmcblk0 | grep -i disk | awk '{print $1}'`	# Name SD
NAMEUSB1=`lsblk | grep -w sda | grep -i disk | awk '{print $1}'`	# Name USB1
NAMEUSB2=`lsblk | grep -w sdb | grep -i disk | awk '{print $1}'`	# Name USB2
NAMEUSB3=`lsblk | grep -w sdc | grep -i disk | awk '{print $1}'`	# Name USB3
NAMEUSB4=`lsblk | grep -w sdd | grep -i disk | awk '{print $1}'`	# Name USB4
DISKS="SD: $NAMESD | USB1: $NAMEUSB1 | USB2: $NAMEUSB2 | USB3: $NAMEUSB3 | USB4: $NAMEUSB4"

# Memory usage
DISK1=`df -h | grep 'dev/root' | awk '{print $2}'` 	# Total
DISK2=`df -h | grep 'dev/root' | awk '{print $3}'` 	# Used
DISK3=`df -h | grep 'dev/root' | awk '{print $4}'`	# Free
CAPACITY="$DISPLAY_DISKTOTAL: $DISK1 | $DISPLAY_DISKUSED: $DISK2 | $DISPLAY_DISKFREE: $DISK3"

# Hostname
HOSTNAME=`hostname -f`

# IP address
if ( /sbin/ifconfig | grep -q "eth0" ) ; then IP_LAN=`ifconfig eth0 | grep "inet" | cut -d ":" -f 2 | awk '{print $2}'` ; else IP_LAN="---" ; fi ; # Lan adress

if ( /sbin/ifconfig | grep -q "wlan0" ) ; then IP_WLAN=`ifconfig wlan0 | grep "inet" | cut -d ":" -f 2 | awk '{print $2}'` ; else IP_WLAN="---" ; fi ;
IP_ADDRESSES="$DISPLAY_IPADDRLAN: $IP_LAN | $DISPLAY_IPADDRWIFI: $IP_WLAN"

# User
USER=`whoami | awk '{print $1}'`

# Last login
LAST1=`last -2 -a | awk 'NR==2{print $3}'`	# Weekday
LAST2=`last -2 -a | awk 'NR==2{print $5}'`	# Day
LAST3=`last -2 -a | awk 'NR==2{print $4}'`	# Month
LAST4=`last -2 -a | awk 'NR==2{print $6}'`	# Time
LAST5=`last -2 -a | awk 'NR==2{print $10}'`	# Remote-Computer
LAST="$LAST1, $LAST2 $LAST3 $LAST4 $DISPLAY_FROM $LAST5"

# Homebridge Version
HOMEBRIDGE=`homebridge --version`

# Node Version
NODE=`node -v | tr -d [:alpha:]`

# Npm Version
NPM=`npm -v | tr -d [:alpha:]`

# Npm Updates
# NPMUPDATES=`npm outdated -g | grep global -i | wc -l`		# Outdated query
NPMUPDATES=`npm-check -g | grep -i " to go from" | wc -l`	# Npm-check query (default)

# Homebridge total
# INSTANZTOTAL="$(ls -l /etc/systemd/system | grep homebridge* | wc -l) $DISPLAY_HOMEBRIDGEINSTANCE"  # Query number of files /etc/systemd/system
INSTANZTOTAL="$(systemctl -t service | grep -w homebridge | wc -l) $DISPLAY_HOMEBRIDGEINSTANCE"		# Query number of systemctl services (default)

# Homebridge active
INSTANZAKTIV="$(systemctl -t service | grep -w homebridge | grep -w active | wc -l) $DISPLAY_HOMEBRIDGEINSTANCE"

# Homebridge inactive
INSTANZINAKTIV="$(systemctl -t service | grep -w homebridge | egrep "activating|failed" | wc -l) $DISPLAY_HOMEBRIDGEINSTANCE"

# deCONZ Version
#DECONZVERSION=`dpkg -s deconz | grep '^Version:' | cut -d "-" -f 1 | awk '{print $2}'`

# deCONZ status
#DECONZSTATUS="$(systemctl -t service | egrep "deconz.service|deconz-gui.service" | grep -w active | wc -l) $DISPLAY_DECONZINSTANCE"


echo -e "\e[1;35m$DISPLAY_TITLE"
echo 
output "yellow"  "$DISPLAY_DATE" "$DATE"											"                   ██                   "
output "magenta" "$DISPLAY_TIME" "$TIME"											"                  ████                  "
output "magenta" "$DISPLAY_MODEL" "$MODEL"											"                ████████    ███████     "
output "magenta" "$DISPLAY_SYSTEM" "$SYSTEM"										"              ███      ██████    ██     "
output "magenta" "$DISPLAY_KERNEL" "$KERNEL"										"            ███           ███    ██     "
output "magenta" "$DISPLAY_UPTIME" "$UPTIME"										"          ███      ███     ██    ██     "
output "magenta" "$DISPLAY_LOAD" "$LOAD"											"        ███      ███ ███    █    ██     "
output "magenta" "$DISPLAY_RAMUSAGE" "$RAM"											"      ███      ███     ███       ██     "
output "magenta" "$DISPLAY_TEMP" "$TEMP"											"    ███      ███         ███      ██    "
output "magenta" "$DISPLAY_DISKS" "$DISKS"											"  ███      ███     ███     ███     ██   "
output "magenta" "$DISPLAY_DISKCAPACITY" "$CAPACITY"								" ███     ███     ███ ███     ███    ██  "
output "magenta" "$DISPLAY_HOSTNAME" "$HOSTNAME"									" ███   ███     ███     ███     ███      "
output "magenta" "$DISPLAY_USERNAME" "$USER"										"   █████     ███         ███    ███     "
output "magenta" "$DISPLAY_LASTLOGIN" "$LAST"										"           ███     ███     ███  ███     "
output "magenta" "$DISPLAY_IPADDR" "$IP_ADDRESSES"									"         ███     ███ ███     ████       "
output "red"     "$DISPLAY_NODEVERSION" "$NODE"										"        ███   ████      ███             "
output "cyan"    "$DISPLAY_NPMVERSION" "$NPM"										"          █████          ███            "
output "cyan"    "$DISPLAY_NPMUPDATES" "$NPMUPDATES $DISPLAY_NPMUPDATESAVAILABLE"	"           ███            ███           "
output "blue"    "$DISPLAY_HOMEBRIDGEVERSION" "$HOMEBRIDGE"							"                         ███            "
output "blue"    "$DISPLAY_HOMEBRIDGETOTAL" "$INSTANZTOTAL"							"                        ███             "
output "blue"    "$DISPLAY_HOMEBRIDGEACTIVE" "$INSTANZAKTIV"						"                       ███              "
output "blue"    "$DISPLAY_HOMEBRIDGEINACTIVE" "$INSTANZINAKTIV"					"                      ███               "
#output "green"   "$DISPLAY_DECONZVERSION" "$DECONZVERSION"							"                    ███                 "
#output "green"   "$DISPLAY_DECONZSTATUS" "$DECONZSTATUS"							"                   ███                  "
